internal/git/git.go

```diff
@@ -118,12 +118,20 @@ func Verify(ctx context.Context, rekor rekor.Verifier, data, sig []byte) (*Verif
 	if err != nil {
 		return nil, fmt.Errorf("error getting signature certs: %w", err)
 	}
+	// TODO: This is very brittle, since the DER encoding will sort the
+	// certificates in the signature, which means that this ordering can
+	// change depending on the issuer.
+	// See https://en.wikipedia.org/wiki/X.690#DER_encoding
+	// We should find a better way to identify the leaf cert that was used
+	// to generate the signature.
+	cert := certs[len(certs)-1]
```

Comment:
To find a leaf certificate, you can inspect the certificates' basic constraints extension. Only the leaf certificate will have `isCa = false`. 

Also in case you're curious, to find the root, it'll be the only certificate where the subject key ID and authority key ID match.

See https://www.pkisolutions.com/basic-constraints-certificate-extension/ for more info

Reply:
I have 2 initial ideas for how to address this

1. we can stick an id in the message attributes
2. we can drop the intermediates and only include the signing cert.

Probably going to go with 2 to start, but we could look into 1 if we want to be fully compliant with `--include-certs`

Reply:
You don't have to store the intermediate and root, as long as gitsign bundles those (which it will, since it's pulling from TUF)

Reply:
Went with 2!